### PR TITLE
Update PelePhysics - Use Sundials submod in gmake

### PR DESCRIPTION
**Pull request by baperry2**
This will fix an error where `make TPL` fails. Mirrors PeleC https://github.com/AMReX-Combustion/PeleC/pull/686

### CODE REVIEW

File: Utils/Make.PeleLMeX [hunk 1]
```diff
@@ -1,7 +1,8 @@
-PELELMEX_HOME ?= /path/to/PeleLMeX
-AMREX_HOME ?= /path/to/AMREX
-PELE_PHYSICS_HOME ?= /path/to/PELE_PHYSICS_HOME
-AMREX_HYDRO_HOME ?= /path/to/AMREX_HYDRO_HOME
+PELELMEX_HOME ?= ../
+AMREX_HOME ?= $(abspath $(lastword $(PELELMEX_HOME)/Submodules/AMReX))
```
Comment (jrood-nrel) [inline Utils/Make.PeleLMeX]: You will need a lowercase `amrex` path.

**Comment by jrood-nrel** [comment]
I think the cloning in the CI files need to clone the submodules now as well

**Comment by jrood-nrel** [comment]
We also need to make the CI more efficient here and have it use CMake soon.

**Comment by baperry2** [comment]
Yeah it would be nice to have the CI match up better between C and LMeX

**Comment by jrood-nrel** [comment]
Looks like `SUNDIALS_HOME` needs to be set in all the other CI files.

**Pull request merged**